### PR TITLE
Added plotly support and minor refactoring of prediction API.

### PR DESCRIPTION
**Pull request by ivanmkc**
- [x] Fixed Dockerfile missing constants.py bug.
- [x] Renamed trainingMethodName to trainingMethodId
- [x] Removed hardcoded prediction parameters.
- [x] Plotly support

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ class SubmitForecastJobAPIRequest(BaseModel):
         prediction_parameters (Dict[str, Any]): Parameters for training.
     """
 
-    trainingMethodName: str
```
Comment (ivanmkc) [inline main.py:127]: Renamed trainingMethodName to trainingMethodId

File: Dockerfile [hunk 2]
```diff
@@ -6,11 +6,12 @@ WORKDIR app
 
 COPY requirements.txt .
 COPY main.py .
+COPY utils.py .
+COPY constants.py .
```
Comment (ivanmkc) [inline Dockerfile:10]: Added missing file.

File: training_methods/bqml_training_method.py [hunk 3]
```diff
@@ -47,19 +81,26 @@ def train(
             str: The model URI
         """
 
-        time_column = model_parameters.get("timeColumn")
-        target_column = model_parameters.get("targetColumn")
-        time_series_id_column = model_parameters.get("timeSeriesIdentifierColumn")
-        forecast_horizon = prediction_parameters.get("forecastHorizon")
+        time_column = model_parameters.get(TIME_COLUMN_PARAMETER)
```
Comment (ivanmkc) [inline training_methods/bqml_training_method.py:84]: Moved inline constants into a constants section at beginning of file.

File: training_methods/debug_training_method.py [hunk 4]
```diff
@@ -55,12 +85,12 @@ def train(
 
         # Sleep for a specified delay, not more than a max.
         delay_in_seconds = min(
-            model_parameters.get("delay_in_seconds", 5), MAX_DELAY_IN_SECONDS
```
Comment (ivanmkc) [inline training_methods/debug_training_method.py:58]: Fixed case.

File: training_methods/training_method.py [hunk 5]
```diff
@@ -25,6 +25,39 @@ def display_name(self) -> str:
         """
         pass
 
+    @abc.abstractmethod
+    def dataset_time_series_identifier_column(
```
Comment (ivanmkc) [inline training_methods/training_method.py:29]: These parameters change depending on the training method, so must be defined by each training method.

File: training_methods/bqml_training_method.py [hunk 6]
```diff
@@ -30,6 +34,36 @@ def display_name(self) -> str:
...
 
+    def dataset_time_series_identifier_column(
+        self, job_request: forecast_job_request.ForecastJobRequest
+    ) -> str:
+        """The column representing the time series identifier variable in the dataset dataframe.
+
+        Returns:
+            str: The column name
+        """
+        return job_request.model_parameters[TIME_SERIES_IDENTIFIER_COLUMN_PARAMETER]
```
Comment (ivanmkc) [inline training_methods/bqml_training_method.py:45]: This changes depending on what the user provides the API.

**Review by soheilazangeneh — APPROVED** [review]

**Pull request merged**